server/entitymanager/entitymanager.go

```diff
@@ -354,3 +354,60 @@ func New(chassisConfigFile string) (*InMemoryEntityManager, error) {
 
 	return newManager, nil
 }
+
+// ReplaceDevice replaces an existing chassis with a new chassis object.
+func (m *InMemoryEntityManager) ReplaceDevice(chassis *service.EntityLookup, newChassis *epb.Chassis) error {
+	// chassis: old device lookup, newChassis: new device
+
+	// todo: validate before replace
+	// todo: forward error from validateConfig
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	delete(m.chassisInventory, *chassis)
+
+	lookup := service.EntityLookup{
+		Manufacturer: newChassis.GetManufacturer(),
+		SerialNumber: newChassis.GetSerialNumber(),
+	}
+
+	m.chassisInventory[lookup] = newChassis
+
+	// will return error when validation is added
```

Comment:
Comments that are complete sentences should be capitalized and punctuated like standard English sentences.

Reply:
I mean... there is no direct object here so we can't call it a complete sentence.

But I got your point, hahaha. I'll fix up the comments. Thanks!

Reply:
Cheers~ Also feel free to resolve conversations once you've addressed the comments :) I'll do it for you now and merge the PR. Thanks Liam~